### PR TITLE
fix(vmsnapshot): snapshotting vm without main network

### PR DESCRIPTION
**Pull request by eofff**
## Description

Fix snapshotting VM without Main network.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: vmsnapshot
type: fix
summary: "Fixed snapshot creation for a virtual machine without the `Main` network."
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Workflow has started.
  Follow the progress here: [Workflow Run](https://github.com/deckhouse/virtualization/actions/runs/24197823657)

  The target step completed with status: **failure**.

**Comment by deckhouse-BOaTswain** [comment]
Workflow has started.
  Follow the progress here: [Workflow Run](https://github.com/deckhouse/virtualization/actions/runs/24214845441)

  The target step completed with status: **failure**.

File: images/virtualization-artifact/pkg/controller/service/restorer/snapshot_resources.go [hunk 1]
```diff
@@ -129,13 +129,19 @@ func (r *SnapshotResources) Prepare(ctx context.Context) error {
 			macAddressNamesByAddress[vmmac.Status.Address] = vmmac.Name
 		}
 
+		hasMainNetwork := len(vm.Spec.Networks) > 0 && vm.Spec.Networks[0].Type == v1alpha2.NetworksTypeMain
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/service/restorer/snapshot_resources.go]: Use, please, `network.HasMainNetworkSpec`

File: images/virtualization-artifact/pkg/controller/vmrestore/internal/life_cycle.go [hunk 2]
```diff
@@ -180,13 +180,18 @@ func (h LifeCycleHandler) Handle(ctx context.Context, vmRestore *v1alpha2.Virtua
 			macAddressNamesByAddress[vmmac.Status.Address] = vmmac.Name
 		}
 
+		hasMainNetwork := len(vm.Spec.Networks) > 0 && vm.Spec.Networks[0].Type == v1alpha2.NetworksTypeMain
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmrestore/internal/life_cycle.go]: Use, please, `network.HasMainNetworkSpec`

File: images/virtualization-artifact/pkg/controller/vmrestore/internal/life_cycle.go [hunk 3]
```diff
@@ -180,13 +180,18 @@ func (h LifeCycleHandler) Handle(ctx context.Context, vmRestore *v1alpha2.Virtua
...
 		for i := range vm.Spec.Networks {
 			ns := &vm.Spec.Networks[i]
 			if ns.Type == v1alpha2.NetworksTypeMain {
 				continue
 			}
 
-			ns.VirtualMachineMACAddressName = macAddressNamesByAddress[macAddressOrder[i-1]]
+			delta := 0
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmrestore/internal/life_cycle.go]: duplication of logic. Let's put it in one place

File: images/virtualization-artifact/pkg/controller/service/restorer/snapshot_resources.go [hunk 4]
```diff
@@ -129,13 +129,19 @@ func (r *SnapshotResources) Prepare(ctx context.Context) error {
...
+
 		for i := range vm.Spec.Networks {
 			ns := &vm.Spec.Networks[i]
 			if ns.Type == v1alpha2.NetworksTypeMain {
 				continue
 			}
 
-			ns.VirtualMachineMACAddressName = macAddressNamesByAddress[macAddressOrder[i-1]]
+			delta := 0
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/service/restorer/snapshot_resources.go]: duplication of logic. Let's put it in one place

**Review by LopatinDmitr — COMMENTED** [review]
The PR is fixing the edge-case, but I haven't seen any tests on it.

**Review by LopatinDmitr — APPROVED** [review]
